src/librustdoc/html/static/js/search.js

```diff
@@ -984,23 +977,22 @@ function initSearch(rawSearchIndex) {
           *                     no match, returns `MAX_LEV_DISTANCE + 1`.
           */
         function checkType(row, elem, literalSearch) {
-            if (row[NAME].length === 0) {
+            if (row.name === null) {
```

Comment:
Maybe I missed it but I only saw you setting `name` to `null` manually and never checking its length. Shouldn't it insert `null` in both "build functions" if the name is empty?